include/seqan/bam_io/bam_index_bai.h

```diff
@@ -154,6 +154,160 @@ class BamIndex<Bai>
+
+    if (refId < 0)
+        return false;  // Cannot seek to invalid reference.
+
+    if (static_cast<unsigned>(refId) >= length(bamIndex._binIndices))
+        return false;  // Cannot seek to invalid reference.
+
+    if (regionStart > regionEnd)
+        return false; // invalid region
+
+    if (regionEnd > 536870912)
+        return false; // BAM format only allows reference of size < 2^29
+
+    // Adjust region
+    // -------------------------------------------------------------------------
+    // We want to mimic samtools view. Since SAM/BAM files are one based we expect
+    // the given region to be also one-based (To be consistent with samtools).
+    // Since SeqAn works 0-based we need to substract 1.
+    SEQAN_ASSERT_GEQ_MSG(regionStart, 1, "Expect 1-based region interval which excludes 0 as a start pos.");
+    --regionStart;
+    --regionEnd;
+
+    // Retrieve the candidate bin identifiers for [regionStart, regionEnd).
+    // -------------------------------------------------------------------------
+    std::vector<uint16_t> candidateBins;
+    _baiReg2bins(candidateBins, regionStart, regionEnd);
+
+    // Retrieve the smallest required offset from the linear index.
+    // -------------------------------------------------------------------------
+    uint64_t linearMinOffset;
+    if (!_getMinFileOffset(linearMinOffset, bamIndex, refId, regionStart))
+        linearMinOffset = 0; // if no linear index query was unsuccessful, value 0 never limits chunk seeking
+
+    // Iterate over chunks in candidate bins and extract records
+    // -------------------------------------------------------------------------
+    BamAlignmentRecord record;
+    for (uint16_t bin : candidateBins)
+    {
+        TMapIter mIt = bamIndex._binIndices[refId].find(bin);
+
+        if (mIt == bamIndex._binIndices[refId].end()) // empty bins are not stored
+            continue;
+
+        for (TChunkIter chunkIt = begin(mIt->second.chunkBegEnds, Rooted()); !atEnd(chunkIt); goNext(chunkIt))
+        {
+            if (chunkIt->i2 < linearMinOffset)
+                continue; // use linearMinOffset to discard unneccessary chunks
+
+            setPosition(bamFile, chunkIt->i1); // jump to candidate chunk
+
+            while(position(bamFile) != static_cast<int>(chunkIt->i2)) // this includes end of file
+            {
+                readRecord(record, bamFile);
+
+                if (record.beginPos > regionEnd)
+                    break; // chunk of consecutive reads is already out of bounds
+
+                // max(1, length) in if clause because if read is unmapped, the endPos calculation is wrong.
+                // (Could be also done ny checking if (unmapped) but might e less efficient than max.
+                if (record.beginPos <= regionEnd &&
+                    static_cast<int>(record.beginPos + std::max(1u, getAlignmentLengthInRef(record)) - 1) >= regionStart)
+                    appendValue(resultContainer, record);
```

Comment:
I am not sure what the default behaviour is, but better, to specify `Generous()` as last parameter.